### PR TITLE
Update listing latest complete & running builds on overview

### LINKED ISSUE #233: Change display of !isIncompleteBuilds on overview
- [x] persist the latest !isIncompleteBuild pipeline and triggered builds on the overview (currently, all builds less than 5 minutes old appear)
- [x] make the persisted !isIncompleteBuild pipeline builds dismissible on the overview

For number 1, [change the filter being applied to the list of builds](https://github.com/openshift/origin-web-console/blob/master/app/scripts/controllers/overview.js#L320)?

Reuse the [existing triggers directive](https://github.com/openshift/origin-web-console/blob/master/app/scripts/directives/resources.js#L81) for number 2?

### PR DESCRIPTION
**Pull request by benjaminapetersen**
- Add methods for sorting builds to BuildService
- update overview.js to use above methods to list 
   - currently running builds
   - the lastest complete build per build config

Fixes #233 

@rhamilto to assist in testing before I remove the [WIP].

### CODE REVIEW

**Comment by rhamilto** [comment]
@benjaminapetersen, you nailed it!  This behaves exactly as expected.  Nice work!

bitmoji

**Comment by benjaminapetersen** [comment]
Ha, +1 emoji :smile:

@jwforres  @spadgett for review!

File: app/scripts/services/builds.js [hunk 1]
```diff
@@ -145,6 +146,61 @@ angular.module("openshiftConsole")
       return moment(endTimestamp).diff(moment(startTimestamp));
     };
 
+    // return only builds that are incomplete
```
Comment (benjaminapetersen) [inline app/scripts/services/builds.js]: Mm... lemme eliminate the comments that tell you exactly the same thing as the function names :confused:

File: app/scripts/services/builds.js [hunk 2]
```diff
@@ -145,6 +146,57 @@ angular.module("openshiftConsole")
...
+
+    var completeBuilds = function(builds) {
+      return _.map(builds, function(build) {
+        return !isIncompleteBuild(build);
+      });
+    };
+
+    var lastCompleteByBuildConfig = function(builds) {
+      return _.reduce(
+                completeBuilds(builds),
```
Comment (jwforres) [inline app/scripts/services/builds.js]: there are far more complete builds than incomplete builds, implementing it this way means you are iterating over almost the entire list of builds in the namespace twice, just in this one function, and this happens anytime there is an update to any build.

**Comment by benjaminapetersen** [comment]
@jwforres updated the `lastCompleteByBuildConfig` function, should be good.

**Comment by jwforres** [comment]
[merge]

**Pull request merged**